frame/staking/src/lib.rs

```diff
@@ -941,6 +941,12 @@ decl_storage! {
 		/// Minimum number of staking participants before emergency conditions are imposed.
 		pub MinimumValidatorCount get(fn minimum_validator_count) config(): u32;
 
+		/// Maximum number of validators participated to produce blocks. 
+		/// 
+		/// If the value `maximum_validator_count > minimum_validator_count` at the end of
+		/// current era `validator_count` will be changed automatically,
+		pub MaximumValidatorCount get(fn maximum_validator_count) config(): u32;
```

Comment:
Im personally not a big fan of combining two behaviors into a variable like this.

Maximum validator count imo should not be used to trigger whether automatic validator selection should occur, but be its own configuration. Happy to have others give their opinion on this though.

If we did combine the two, at least i would pick a less generic name, like "AutoAdjustValidatorCountMax", but hopefully you may think of a cleaner name that than.